src/pipecat/services/playht.py

```diff
@@ -30,36 +40,281 @@
+                else:
+                    raise Exception(f"Failed to get WebSocket URL: {response.status}")
+
+    async def _disconnect(self):
+        if self._websocket:
+            await self._websocket.close()
+            self._websocket = None
+
+    async def _receive_task_handler(
+        self, websocket: websockets.WebSocketClientProtocol
+    ) -> AsyncGenerator[Tuple[bytes, bool], None]:
+        audio_chunks = []
+        end_of_stream = False
+        header_size = 78  # Size of the WAV header which we'll skip
+        header_received = False
+
+        while not end_of_stream:
+            try:
+                message = await asyncio.wait_for(websocket.recv(), timeout=10.0)
+                if isinstance(message, bytes):
+                    chunk_size = len(message)
+
+                    # Skip the WAV header
+                    if not header_received and chunk_size == header_size:
+                        header_received = True
+                        continue
+
+                    audio_chunks.append(message)
+                    yield message, False
+                else:
+                    logger.debug(f"Received text message: {message}")
+                    try:
+                        msg = json.loads(message)
+                        if "request_id" in msg:
+                            logger.debug("End of audio stream")
+                            end_of_stream = True
+                    except json.JSONDecodeError:
+                        logger.error(f"Invalid JSON message: {message}")
+            except asyncio.TimeoutError:
+                logger.warning("Timeout waiting for WebSocket message")
+                break
+
+    async def run_tts(self, text: str) -> AsyncGenerator[Frame, None]:
+        logger.debug(f"Generating TTS: [{text}]")
+
+        try:
+            if not self._websocket_url:
+                await self._get_websocket_url()
+
+            if not isinstance(self._websocket_url, str):
+                raise ValueError("WebSocket URL is not a string")
+
+            async with websockets.connect(self._websocket_url) as websocket:
+                self._websocket = websocket
+
+                await self.start_ttfb_metrics()
+                yield TTSStartedFrame()
+
+                tts_command = {
+                    "text": text,
+                    "voice": self._voice_id,
+                    "voice_engine": self._settings["voice_engine"],
+                    "output_format": self._settings["output_format"],
+                    "sample_rate": self._settings["sample_rate"],
+                    "language": self._settings["language"],
+                    "speed": self._settings["speed"],
+                    "seed": self._settings["seed"],
+                }
+
+                await websocket.send(json.dumps(tts_command))
+
+                await self.start_tts_usage_metrics(text)
+
+                async for audio_chunk, _ in self._receive_task_handler(websocket):
```

Comment:
Is we can keep the websocket open we just need to create this task once, instead of every time.